packages/zod-rpc/readme.md

```diff
@@ -0,0 +1,244 @@
+# zod-rpc
+
+## RPC calls
+
+### 1 Create a new RPC call that **requires** a response
+
+In any file, create a class with the following structure, note that the `validator` here would
+be located in a sibling file with the `.zod.js` extension. This is how we are able to strip validation
+for the production builds.
+
+```js
+import { ZodRPC } from "../packages/zod-rpc";
+
+// These need to be in a separate file with the *.zod.js extension
+const paramsValidator = z.object({name: z.string()});
+const resultValidator = z.object({name: z.string()});
+
+/** @extends {ZodRPC<paramsValidator, resultValidator>} */
+class Login extends ZodRPC {
+    method = "loginUser"
+    id = "loginUserResponse"
+    paramsValidator = paramsValidator
+    resultValidator = resultValidator
+}
+
+const result = await ioHandler.request(new Login({name: "dax"}))
+```
+
+Explanation:
+* `/** @extends {ZodRPC<paramsValidator, resultValidator>} */` is how we can forward generic parameters when using JSDoc with Typescript 
+* `class Login extends ZodRPC` creates a class `Logic` that inherits all the functionality of a ZodRPC call (validation)
+* `method = "loginUser"` depending on the platform in question this may be used in various ways
+*   - For example, on macOS/iOS the `method` is used to lookup a specific named handler that the native side injects
+*   - But on Windows, the `method` is sent via a generic handler such as `postMessage({type: 'Autofill::' + rpc.method})`
+* `paramsValidator` this is the Zod Validator used to validate the `params` (outgoing) data
+* `resultValidator` this is the Zod Validator used to validate the `result` (incoming) data
+ 
+Please see **Send a request** section for details on how to actually send this  
+
+### 2 Create a new RPC call that does NOT require a response
+
+This is the same process as above, except you do not need to specify an `id` or `resultValidator`.
+
+```javascript
+const paramsValidator = z.object({name: z.string()});
+
+/** @extends {ZodRPC<paramsValidator>} */
+class Login extends ZodRPC {
+    method = "loginUser"
+    paramsValidator = paramsValidator
+}
+
+ioHandler.notify(new Login({name: "dax"}))
+```
+
+
+### 3 Create an RPC call from a name + data only 
+
+In situations where you just want to quickly create an RPC call, but you don't care/need to have a 
+class definition (eg: because you're accepting that you won't be able to use `instanceof` checks late, then 
+you can do the following:
```

Comment:
Can you think of an example of when we might prefer not to use a class definition? Why would or wouldn't I check for `instanceof`? What are the pros/cons?